game/scripts/vscripts/items/farming/modifier_passive_gpm.lua

```diff
@@ -19,14 +19,19 @@ function modifier_passive_gpm:GetTexture()
 end
 
 if IsServer() then
-  function modifier_passive_gpm:OnIntervalThink ()
+  function modifier_passive_gpm:OnIntervalThink()
     if not PlayerResource then
       -- sometimes for no reason the player resource isn't there, usually only at the start of games in tools mode
       return
     end
     local caster = self:GetCaster()
     local gpm = self:GetAbility():GetSpecialValueFor('bonus_gold_per_minute')
-    Gold:ModifyGold(caster:GetPlayerID(), gpm / 60, true, DOTA_ModifyGold_GameTick)
+
+    -- Don't give gold on illusions or Meepo clones
+    if caster:IsIllusion() or caster:IsClone() then
```

Comment:
Should you check if they're a tempest double too?

Reply:
It's been a while. Forgot that Tempest Double requires a separate check. Good catch.